pandas/core/categorical.py

```diff
@@ -1340,8 +1337,12 @@ def _repr_categories_info(self):
         """ Returns a string representation of the footer."""
 
         category_strs = self._repr_categories()
-        levheader = "Categories (%d, %s): " % (len(self.categories),
-                                               self.categories.dtype)
+        if self.categories.inferred_type == 'period':
```

Comment:
ok, I think to the ``Index`` add a property that returns ``.dtype`` (and returns ``.inferred_type`` for a ``PeriodIndex``), maybe ``getattr(self.categories,'dtype_str',self.categories.dtype)``; this will also allow it to word if ``categories`` are an ``ndarray``, though I don't think we actually allow this.